### ISSUE TITLE
.request-signature__footer__sign-button not found

### ISSUE THREAD

**Issue by teejteej** [ChainSafe/dappeteer#89]
**Describe the bug**
Everything worked fine two weeks ago, but since a few days it starts to hang on the signing process. Here is the error:
```TimeoutError: waiting for selector `.request-signature__footer__sign-button` failed: timeout 30000ms exceeded```

**System:**
- OS: OS-X
- NodeJs version v17.3.0
- dAppeteer version 2.3.0
- Metamask version v10.1.1

**Comment by keithfrazier98** [comment]
I use this function for my approvals:
```
export async function approveTransactions(metamask: dappeteer.Dappeteer, txAmount: number) {
  //open MM
  await metamask.page.bringToFront();
  const activity = await useXPath(metamask.page, "button", "Activity", false);
  //@ts-ignore
  await activity.click();
  for (let tx = 0; tx < txAmount; tx++) {
    //wait for second tx, click and confirm
    try {
      await clearMMPopup(metamask);
      await metamask.page.waitForSelector(".home__container");
      await metamask.page.waitForSelector("li[data-testid=home__activity-tab] > button");
      await metamask.page.click("li[data-testid=home__activity-tab] > button");
      await metamask.page.waitForSelector(".transaction-status.transaction-status--unapproved");
      await metamask.page.reload();
      await metamask.page.waitForSelector(".btn-primary");
      await metamask.confirmTransaction();
    } catch (error) {
      console.log(error);
    }
  }
  return;
}
```

**Comment by robinicus** [comment]
Thank you for posting that @keithfrazier98 and @teejteej  - I am struggling with this exact issue of the sign() function not working at all currently as, after submitting the form on the site, the site takes a few seconds to make the request back for a signature and, in that time, the dapeteer sign() function has already switched back to the metamask tab - when that happens, the original site detects that the tab has been switched off and doesn't make the sign request. 

Can I ask where you are getting useXPath from in your code above? Sorry for the somewhat simple question however I get a ReferenceError: useXPath is not defined. Are you importing nightwatch somewhere as well as dappeteer?

to do it in JS rather than TS, I changed the function to:

async function approveTransactions(metamask, txAmount) 

and am calling it with:

await approveTransactions(metamask, 1);

after the form is submitted on the site.

**Comment by keithfrazier98** [comment]
@robinicus they should all work if you just remove the type declarations on the arguments. I use [confirmTransaction()](https://github.com/ChainSafe/dappeteer/blob/master/docs/API.md#confirmTransaction)  to sign a signature request, I haven't actually found a use case for the sign() function.  Also, about what you said here:

> the original site detects that the tab has been switched off and doesn't make the sign request

Make sure that is your actual issue because I ran into an issue similar to this. In my case, upon sending the signature request, puppeteer or dappateer (not sure which one) is exiting the metamask pop up so fast I wasn't noticing. When the metamask popup is closed it actually rejects the transaction and by the time dappeteer opens the separate metamask page, there is nothing to sign.

**Comment by robinicus** [comment]
Thanks for your comments @keithfrazier98 you finally got me over the line and it's working! I will test to see how robust it is and also how I can throw and flag an error if it does not work so I can then re-do the transaction as you've mentioned.

**Comment by mpetrunic** [comment]
should be fixed in 3.0.0-rc.0
